File: DONGUK/프로그래머스_WEEK 1/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    rank = {6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6}
    high, low = 1, 6
    
    candidate = []
    wrong_cnt = 0
    for num in lottos:
        if num not in win_nums:
            candidate.append(num)
            
            if num != 0:
                wrong_cnt += 1
    
            
    N = len(candidate)
    if N == 0: # 다 맞는 경우
        high, low = 1, 1
        answer.append(high)
        answer.append(low)
    
    else:
        answer.append(rank[6-wrong_cnt])
        answer.append(rank[6-N])
    
    return answer

File: JINSEOP/프로그래머스_지난_문제/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

unordered_map<int, int> rank_caculator;

void init_calculator(vector<int> lottos, vector<int> win_nums) {
    for (int i = 0; i < 6; i++) {
        rank_caculator[lottos[i]]++;
        rank_caculator[win_nums[i]]++;
    }
}

void calc_rank(vector<int>& answer) {
    int rank = 7, zero_num = 0;

    for (auto comp : rank_caculator) {
        if (comp.first != 0 && comp.second > 1)
            rank--;
        if (comp.first == 0)
            zero_num = comp.second;
    }

    answer.push_back(((rank - zero_num) > 6 ? 6 : (rank - zero_num)));
    answer.push_back(rank > 6 ? 6 : rank);
}

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;

    init_calculator(lottos, win_nums);
    calc_rank(answer);

    return answer;
}
